pkg/forwarder/domain_forwarder.go

```diff
@@ -36,36 +35,39 @@ type domainForwarder struct {
 	retryQueue                   []Transaction
```

Comment:
`retryQueue` is still require when using `forwarder_retry_queue_max_size`.
Ideally both logics should be merged. I didn't do it because:
 - A lack of time as it makes the dev more complex
 - I would like to minimize the code change in the domain_forwarder to avoid regression.
 - Keeping the exact same behavior of the forwarder is tricky. 

I think it is something that can be done as a second steps if it is required.

Reply:
OK, I understand and agree with the reasons. But I feel we should do it. I'm afraid we're never going to come back to it if we don't do it now. Also if we're going add change in behavior it's better to do it along side the new feature that explain the change.

Reply:
I do agree.